pageserver/src/thread_mgr.rs

```diff
@@ -126,7 +126,7 @@ struct PageServerThread {
 
     /// Handle for waiting for the thread to exit. It can be None, if the
     /// the thread has already exited.
-    join_handle: Mutex<Option<JoinHandle<()>>>,
+    join_handle: Option<JoinHandle<()>>,
```

Comment:
With the new Mutex, `shutdown_requested` also doesn't need to be atomic anymore.

Alternatively, you could add the new Mutex just over tenant and timeline fields. Or use a single mutex for tenant, timeline and join_handle. That would make it more clear which fields can change after creation. And would avoid acquiring the lock in is_shutdown_requested, though I don't think it makes any measurable performance difference in practice.